File: server/dotnet/Payment/CreatePaymentController.cs
using Microsoft.AspNetCore.Mvc;
using OnlinePayments.Sdk.Domain;

namespace OnlinePayments.Sdk.Example;

[Route("api/createpayment")]
[ApiController]
public class CreatePaymentController
{
    private readonly CreatePaymentMapper _createPaymentMapper;
    private readonly CreatePaymentService _createPaymentService;
    private readonly PaymentDetailsService _paymentDetailsService;

    public CreatePaymentController(
        CreatePaymentMapper createPaymentMapper,
        CreatePaymentService createPaymentService,
        PaymentDetailsService paymentDetailsService
    )
    {
        _createPaymentMapper = createPaymentMapper;
        _createPaymentService = createPaymentService;
        _paymentDetailsService = paymentDetailsService;
    }

    [HttpGet]
    public CreatePaymentBasicDto InitializeRequest() {
        return _createPaymentMapper.ToEmptyBasicDto();
    }

    [HttpGet("outcome")]
    public async Task<PaymentDetailsResponse> GetPaymentResponse(
        [FromQuery] string paymentId
    ) {
        if (paymentId == null)
            throw new Exception("Payment id is missing!");

        return await _paymentDetailsService.GetPaymentDetails(paymentId);
    }

    [HttpPost("basic")]
    public async Task<CreatePaymentResponse> CreatePaymentResponse(
        [FromBody] CreatePaymentBasicDto createPaymentBasicDto
    ) {
        return await _createPaymentService.CreatePayment(
                _createPaymentMapper.ToCreatePaymentRequest(createPaymentBasicDto));
    }

}

File: server/nodejs/src/payment/createPaymentController.js
const mapper = require('./createPaymentMapper')
const paymentDetailsService = require('./paymentDetailsService')
const createPaymentService = require('./createPaymentService')

/**
 * Initializes a request with an empty basic create payment dto
 */
exports.initializeRequest = () => {
    return mapper.toEmptyBasicDto();
}

/**
 * Gets a payment response for a provided payment id as a query param
 */
exports.getPaymentResponse = async function (req) {
    if (req.query.paymentId === undefined || req.query.paymentId === '')
        throw new Exception("Payment id is missing");

    const paymentId = req.query.paymentId;    

    const paymentDetailsResponse = await paymentDetailsService.getPaymentDetails(paymentId);
    return paymentDetailsResponse.body
}

/**
 * Performs a basic payment request
 */
exports.createPaymentRequest = async function (req) {
    const paymentRequest = mapper.toPaymentRequest(req.body);
    const paymentResponse = await createPaymentService.createPayment(paymentRequest)
    return paymentResponse.body
}
